Translate getting started (#126)
reactjs/pt-BR.reactjs.org: content/docs/getting-started.md

  - "docs/environments.html"
---

> Esta página é um visão geral da documentação do React e de recursos relacionados.

Reviewer: ```suggestion
Esta página é um visão geral da documentação do React e dos recursos relacionados.
```

**React** é uma biblioteca JavaScript para construção de interfaces de usuário. Saiba tudo sobre React em [nossa página inicial](/) ou [através do tutorial](/tutorial/tutorial.html).

---

- [Pratique React](#try-react)
- [Aprenda React](#learn-react)
- [Mantenha-se informado](#staying-informed)

> - [Documentação das versões](#versioned-documentation)

Reviewer 2: ```suggestion
- [Documentação Versionada](#versioned-documentation)
```

Author: Não acha que `das versões` fica mais claro aqui não? Para mim facilita o entendimento.

Author: Para deixar mais claro aqui o meu entendimento, para mim a versão da documentação está fortemente ligada à versão da lib. Então, penso que `documentação versionada` leva a entender que a documentação tem uma versão isolada da biblioteca, quando não é.

Author: Como se tívessemos uma documentação v1, v2 enquanto a lib é v16.6, v16.7.

Reviewer 3: Concordo

- [Esquecemos algo?](#something-missing)

## Pratique React {#try-react}

> React foi pensado desde o início para ser adotado gradualmente e **você pode usar mais ou menos de acordo com sua necessidade.** Se você quer uma amostra do React, adicionar interatividade a uma página HTML simples ou iniciar uma aplicação React complexa, os links desta seção o ajudarão no início.

Reviewer 2: ```suggestion
React foi pensado desde o início para ser adotado gradualmente e **você pode usar menos ou mais da biblioteca de acordo com sua necessidade.** Se você quer uma amostra do React, adicionar interatividade a uma página HTML simples ou iniciar uma aplicação React complexa, os links desta seção o ajudarão no início.
```
Só pq "mais ou menos" naquela construção estava dando uma sensação de que você poderia usar o React "marromeno", como se fosse algo no meio termo. A

cho que "menos ou mais" por não ser uma expressão que já traz um significado, fica melhor para expressar o sentido original.

### Editores Online {#online-playgrounds}

Se você está interessado em testar um pouco o React, pode usar um editor de código online. Tente um Olá Mundo no [CodePen](codepen://hello-world) ou [CodeSandbox](https://codesandbox.io/s/new).

Se prefere usar o seu próprio editor de texto, pode também [baixar este arquivo HTML](https://raw.githubusercontent.com/reactjs/reactjs.org/master/static/html/single-file-example.html), editá-lo e a partir do seu sistema de arquivos abrir em seu navegador web. Este arquivo faz uma lenta transformação de código em tempo de execução, portanto nós recomendamos usar somente para demonstrações simples.

### Adicione React a um Website {#add-react-to-a-website}

Você pode [adicionar React a uma página HTML em um minuto](/docs/add-react-to-a-website.html). Você pode então expandir gradualmente seu uso ou mantê-lo limitado a alguns widgets dinâmicos.

### Crie uma nova Aplicação React {#create-a-new-react-app}

Ao iniciar um projeto React, [uma simples página HTML com tags script](/docs/add-react-to-a-website.html) pode ser a melhor opção. É necessário apenas um minuto para configurar!

> A medida que sua aplicação cresce, você pode levar em consideração uma configuração mais integrada. Há [inúmeras ferramentas JavaScript](/docs/create-a-new-react-app.html) que nós recomendamos para aplicações grandes. Cada uma delas exige pouca ou até nenhuma configuração e permite a você aproveitar ao máximo o rico ecossistema do React.

Reviewer 2: ```suggestion
A medida que sua aplicação cresce, você pode começar a considerar uma configuração mais integrada. Há [inúmeras ferramentas JavaScript](/docs/create-a-new-react-app.html) que nós recomendamos para aplicações grandes. Cada uma delas exige pouca ou até nenhuma configuração e permite a você aproveitar ao máximo o rico ecossistema do React.
```

Acho que "levar em consideração" aqui cria um sentido um pouco diferente do original. 

Ainda que o sentido de "considerar" a nova solução seja o mesmo, eu creio que a omissão do "querer" (want)  remove um pouco o sentido que era o de: "sua aplicação vai começar a te dar trabalho se vc não integrar direito, então vc definitivamente vai querer começar a considerar algo mais integrado"

## Aprenda React {#learn-react}

As pessoas chegam ao React com diferentes conhecimentos e com diferentes métodos de aprendizado. Independente de preferir uma abordagem mais teórica ou mais prática, esperamos que ache esta seção útil.

* Se você preferir **aprender fazendo**, comece com o nosso [tutorial prático](/tutorial/tutorial.html).
* Se você preferir **aprender os conceitos passo-a-passo**, inicie com o nosso [guia dos principais conceitos](/docs/hello-world.html).

> Como qualquer tecnologia desconhecida, React tem uma curva de aprendizado. Com prática e um pouco de paciência, você *pegará* o jeito.

Reviewer 2: Massa! Ficou muito bom!

### Primeiros exemplos {#first-examples}

A [página inicial do React](/) contém alguns pequenos exemplos de React com um editor incluso. Mesmo que você ainda não saiba nada sobre o React, tente mudar o código do editor e veja como isso afetará o resultado.

### React para Iniciantes {#react-for-beginners}

Se você acha que a documentação do React vai em um ritmo mais rápido do que você está acostumado, confira [este resumo do React por Tania Rascia](https://www.taniarascia.com/getting-started-with-react/). Ele introduz os conceitos React mais importantes de maneira detalhada e amigável a quem é iniciante. Assim que estiver pronto, experimente a documentação novamente!

### React para Designers {#react-for-designers}

Se você vem da área de design, [estes recursos](http://reactfordesigners.com/) são um ótimo lugar para começar.

### Recursos JavaScript {#javascript-resources}

> A documentação do React assume que você tem alguma familiaridade com a programação na linguagem JavaScript. Você não precisa ser um especialista, mas é mais difícil aprender o React e o JavaScript ao mesmo tempo.

Reviewer: ```suggestion
A documentação do React assume que você tem alguma familiaridade com a programação na linguagem JavaScript. Você não precisa ser um especialista. Porém, é mais difícil aprender o React e o JavaScript ao mesmo tempo.
```

Author: @glaucia86 Nesse caso, não acha que a sílaba faz mais sentido? Pois a conjunção (tanto faz usar `mas` como `porém`) está sendo usada para oposição a ideia da primeira oração.

Reviewer 2: @lucianomlima 

Concordo com a @glaucia86 nesse ponto, apesar de dar o mesmo sentido e estar gramaticalmente correto, "mas é mais" fica um pouco cacofônico. Usar o "porém" deixa a leitura mais agradável.

Author: A dúvida não foi sobre o `porém` e sim a troca da vírgula pelo ponto. Ali é uma oração composta onde a segunda é uma antítese da primeira.

> Recomendamos que você leia [esta visão geral do JavaScript](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/A_re-introduction_to_JavaScript) para verificar seu nível de conhecimento. Ela levará entre 30 minutos e uma hora, mas você se sentirá mais confiante para aprender Ract.

Reviewer: ```suggestion
Recomendamos que você leia [esta visão geral do JavaScript](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/A_re-introduction_to_JavaScript) para verificar seu nível de conhecimento. Ela levará entre 30 minutos e uma hora. Assim, você se sentirá mais confiante para aprender React.
```

>Dica
>

> >Sempre que você se sentir confuso com algo em JavaScript, o [MDN](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript) e o [javascript.info](http://javascript.info/) serão ótimos sites para verificar. Há também [fóruns de suporte da comunidade](/community/support.html) onde você poderá pedir ajuda.

Reviewer: ```suggestion
Sempre que você se sentir confuso com algo em JavaScript, o [MDN](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript) e o [javascript.info](http://javascript.info/) serão ótimos sites para conferir. Há também [fóruns de suporte da comunidade](/community/support.html) onde você poderá pedir ajuda.
```

> >Sempre que você se sentir confuso com algo em JavaScript, o [MDN](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript) e o [javascript.info](http://javascript.info/) serão ótimos sites para conferir. Há também [fóruns de suporte da comunidade](/community/support.html) onde você poderá pedir ajuda.

Reviewer 2: ```suggestion
>Sempre que você se sentir confuso com algo em JavaScript, o [MDN](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript) e o [javascript.info](https://javascript.info/) serão ótimos sites para conferir. Há também [fóruns de suporte da comunidade](/community/support.html) onde você poderá pedir ajuda.
```

HTTPS nas urls, pra evitar conflitos no merge com o core.

Author: Passou desapercebido! Thanks!

### Tutorial Prático {#practical-tutorial}

> Se você prefere **aprender fazendo**, confira nosso [tutorial prático](/tutorial/tutorial.html). Neste tutorial, criamos um jogo-da-velha no React. Você pode pensar em evitá-lo porque não está construindo jogos – mas dê uma chance. As técnicas que você aprenderá no tutorial são fundamentais para criar *qualquer* aplicação React e dominá-las dará a você uma compreensão muito mais profunda.

Reviewer: ```suggestion
Se você prefere **aprender fazendo**, confira nosso [tutorial prático](/tutorial/tutorial.html). Neste tutorial, criamos um jogo-da-velha em React. Você pode pensar em evitar desenvolver porque não está construindo jogos – mas dê uma chance. As técnicas que você aprenderá no tutorial são fundamentais para criar *qualquer* aplicação React e ao dominá-las dará a você uma compreensão muito mais profunda.
```

### Guia Passo-a-passo {#step-by-step-guide}

Se você prefere **aprender conceitos passo-a-passo**, nosso [guia dos principais conceitos](/docs/hello-world.html) é o melhor lugar para começar. Cada próximo capítulo é baseado no conhecimento apresentado nos capítulos anteriores, para que você não perca nada ao longo do processo.

### Pensando em React {#thinking-in-react}

> Muitos usuários de React creditam à leitura [Pensando em React](/docs/thinking-in-react.html) como o momento em que React finalmente "fez sentido" para eles. É provavelmente o passo-a-passo mais antigo do React, mas ainda é relevante.

Reviewer: ```suggestion
Muitos usuários do React dão créditos a leitura [Pensando em React](/docs/thinking-in-react.html) como o momento em que React finalmente "fez sentido" para eles. É provavelmente o passo-a-passo mais antigo do React.,mas que ainda é relevante.
```

Author: Em `dão créditos a leitura...` é `a` craseado. Pode ser substituído por `dão créditos para a leitura...`.

### Cursos Recomendados {#recommended-courses}

> As vezes, as pessoas acham que livros e cursos em vídeo de terceiros são mais úteis do que a documentação oficial. Mantemos [uma lista de recursos recomendados](/community/courses.html), alguns dos quais são gratuitos.

Reviewer: ```suggestion
Às vezes, as pessoas acham que livros e cursos em vídeo de terceiros são mais úteis do que a documentação oficial. Nós mantemos [uma lista de recursos recomendados](/community/courses.html), alguns dos quais são gratuitos.
```

### Conceitos Avançados {#advanced-concepts}

> Uma vez que você esteja confortável com os [conceitos principais](#main-concepts) e brincou um pouco com o React, pode se interessar em tópicos mais avançados. Esta seção apresentará poderosos, porém pouco utilizados, recursos do React, como [context](/docs/context.html) e [refs](/docs/refs-and-the-dom.html).

Reviewer: ```suggestion
Uma vez que você esteja confortável com os [conceitos principais](#main-concepts) e brincou um pouco com o React, talvez você pode se interessar em tópicos mais avançados. Esta seção apresentará poderosos, porém pouco utilizados, recursos do React, como [context](/docs/context.html) e [refs](/docs/refs-and-the-dom.html).
```

> Uma vez que você esteja confortável com os [conceitos principais](#main-concepts) e brincou um pouco com o React, talvez você possa se interessar em tópicos mais avançados. Esta seção apresentará poderosos, porém pouco utilizados, recursos do React, como [context](/docs/context.html) e [refs](/docs/refs-and-the-dom.html).

Reviewer 2: ```suggestion
Uma vez que você esteja confortável com os [conceitos principais](#main-concepts) e brincou um pouco com o React, talvez você possa se interessar em tópicos mais avançados. Esta seção apresentará poderosos, porém menos utilizados, recursos do React, como [context](/docs/context.html) e [refs](/docs/refs-and-the-dom.html).
```

Acho que aqui o sentido é de comparativo. Elas são **menos** utilizadas que as demais.

### Referência da API {#api-reference}

> Esta seção da documentação será útil quando você quiser saber mais sobre uma API React em particular. Por exemplo, [a referência da API do `React.Component`](/docs/react-component.html) pode te dar detalhes de como funciona o `setState()` e quais métodos do ciclo de vida são úteis para o mesmo.

Reviewer: ```suggestion
Esta seção da documentação será útil quando você quiser saber mais sobre uma determinada API em particular do React. Por exemplo, [a referência da API do `React.Component`](/docs/react-component.html) pode te dar detalhes de como funciona o `setState()` e quais os métodos do ciclo de vida são úteis para o mesmo.
```

Author: Determinada e particular são sinônimos nesse contexto. É realmente necessário?

Reviewer 2: ```suggestion
Esta seção da documentação será útil quando você quiser saber mais sobre uma API do React específica. Por exemplo, [a referência da API do `React.Component`](/docs/react-component.html) pode te dar detalhes de como funciona o `setState()` e quais métodos do ciclo de vida são úteis para o mesmo.
```

Creio que "específica" seria o mais adequado e menos literal. Acho que "em particular" apesar de correto, não é muito usual para denotar especificidade.

Já no final, creio que o "o" é opcional.

### Glossário e FAQ {#glossary-and-faq}

O [glossário](/docs/glossary.html) contém uma visão geral dos principais termos que você verá na documentação do React. Há também uma seção de FAQ dedicada a perguntas e respostas sobre assuntos comuns, incluindo [requisições AJAX](/docs/faq-ajax.html), [estado do componente](/docs/faq-state.html) e [estrutura de arquivos](/docs/faq-structure.html).

## Mantenha-se Informado {#staying-informed}

O [blog do React](/blog/) é a fonte oficial para as atualizações feitas pelo time do React. Qualquer coisa importante, incluindo notas de lançamento ou notificações de depreciação, serão postadas primeiro lá.

Você também pode seguir o [@reactjs](https://twitter.com/reactjs) no Twitter, mas você não perderá nenhuma informação essencial se apenas ler o blog.

> Nem toda versão do React tem sua própria publicação no blog, mas você encontrará um detalhado registro de mudanças para cada versão [no arquivo `CHANGELOG.md` no repositório do React](https://github.com/facebook/react/blob/master/CHANGELOG.md), assim como na página de [Versões](https://github.com/facebook/react).

Reviewer: ```suggestion
Nem toda versão do React tem sua própria publicação no blog. Mas, você encontrará um detalhado registro de mudanças para cada versão [no arquivo `CHANGELOG.md` no repositório do React](https://github.com/facebook/react/blob/master/CHANGELOG.md), assim como na página de [Versões](https://github.com/facebook/react).
```

> ## Documentação das Versões {#versioned-documentation}

Reviewer 2: ```suggestion
## Documentação Versionada {#versioned-documentation}
```

Creio que aqui ele está especificando o fato da documentação ser versionada.

Reviewer: @lucianomlima oq ficou resolvido aqui? 

Versioned => Versionado(a). Chegou a fazer a alteração?

Author: @glaucia86 estou aguardando só algumas respostas nas conversations abertas. Tendo elas respondidas, mando o último ajuste!

Reviewer: @halian-vilela lendo o contexto abaixo... a tradução mais lógica seria: #Documentação das Versões. 
Acho q nesse caso seria válido manter dessa maneira. Senão... abaixo... teremos que alterar para documentação versionada (onde aparece versões). 
Aprovando essa PR

> Esta documentação sempre refletirá a última versão estável do React. Desde o React 16, você pode encontrar a documentação das versões antigas [em uma página separada](/versions). Note que a documentação das versões antigas foi criada no momento de seu lançamento e por isso podem não estar devidamente atualizadas.

Reviewer 2: ```suggestion
Esta documentação sempre refletirá a última versão estável do React. Desde o React 16, você pode encontrar a documentação das versões antigas [em uma página separada](/versions). Note que as documentações das versões antigas ficam registradas no momento de seu lançamento e por isso não ficam sendo continuamente atualizadas.
```

> ## Esquecemos algo? {#something-missing}

Reviewer 3: Acho que aqui poderia ser mantido o "TitleCase" em **A**lgo.

Author: Done!
